src/main/java/seedu/duke/commands/DeleteCommand.java

```diff
@@ -47,7 +47,7 @@ public DeleteCommand(int taskIndex, String taskModule) {
 
     @Override
     public CommandResult execute(ModuleList moduleList, Configuration configuration) throws ModHappyException {
-        if (taskIndex < 0) { //If invalid task number or no task number was provided
+        if (!Objects.isNull(moduleCode)) {
             deleteModule(moduleList);
```

Comment:
I suggest keep the comments.

Reply:
The logic is different now. Last time it was "when invalid task number or no task number was provided by the user, we assume that it is not `del task` but rather it's a `del mod [MODULE_CODE]` input and thus execute the `deleteModule(moduleList)`. Now, it is "execute deleteModule when moduleCode is detected", which I think makes more sense. Also, now the invalid task number/no task number errors have been captured as exceptions when parsing, so it shouldnt appear in DeleteCommand.